org.eclipse.scanning.event.ui/src/org/eclipse/scanning/event/ui/view/StatusQueueView.java

```diff
@@ -289,22 +256,20 @@ public void dispose() {
 	 *
 	 * @param bean
 	 */
-	protected void mergeBean(final StatusBean bean) throws Exception {
+	protected void mergeBean(final StatusBean bean) {
 
-		getSite().getShell().getDisplay().asyncExec(new Runnable() {
-			@Override
-			public void run(){
+		getSite().getShell().getDisplay().asyncExec(() -> {
 				if (queue.containsKey(bean.getUniqueId())) {
 					queue.get(bean.getUniqueId()).merge(bean);
 					viewer.refresh();
 					updateSelected();
 				} else {
 					reconnect();
 				}
-			}
-		});
+			});
 	}
 
+	@SuppressWarnings("squid:S3776")
 	private void createActions() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)